From fc029f49e23d01c0be572b199790e8747a1cba0c Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Thu, 9 Mar 2023 11:56:06 +0100
Subject: [PATCH 065/120] remoteproc: tee: add firmware ID parameter on
 open_session.

To be able to retrieve instance associated to a firmware,
Add the firmware ID as parameter on open_session.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git 9681673f35431854ecf806d805f9c1e8ba89ec1f remoteproc: tee: add firmware ID parameter on open_session.]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I3a16efa78f19ce09f441612fb4bb124e3c755e6a
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/315248
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Michel JAOUEN <michel.jaouen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
---
 drivers/remoteproc/rproc-optee.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/remoteproc/rproc-optee.c b/drivers/remoteproc/rproc-optee.c
index 4adc568773b..d0e08bd560b 100644
--- a/drivers/remoteproc/rproc-optee.c
+++ b/drivers/remoteproc/rproc-optee.c
@@ -193,6 +193,7 @@ int rproc_optee_open(struct rproc_optee *trproc)
 	struct udevice *tee = NULL;
 	const struct tee_optee_ta_uuid uuid = TA_REMOTEPROC_UUID;
 	struct tee_open_session_arg arg = { };
+	struct tee_param param;
 	int rc;
 
 	if (!trproc)
@@ -203,7 +204,11 @@ int rproc_optee_open(struct rproc_optee *trproc)
 		return -ENODEV;
 
 	tee_optee_ta_uuid_to_octets(arg.uuid, &uuid);
-	rc = tee_open_session(tee, &arg, 0, NULL);
+
+	param.attr = TEE_PARAM_ATTR_TYPE_VALUE_INPUT;
+	param.u.value.a = trproc->fw_id;
+
+	rc = tee_open_session(tee, &arg, 1, &param);
 	if (rc < 0 || arg.ret != 0) {
 		if (!rc)
 			rc = -EIO;
-- 
2.47.2

